run_operations.py

```diff
@@ -119,168 +128,181 @@ def command_line_parse(args):
+
+		Parameters: none
+		Returns:    ssara_options: [str], array of command line options to run ssara_federated_query with
+	"""
+
+	with open('/nethome/' + user + '/insarlab/OPERATIONS/TEMPLATES/' + dataset + '.template', 'r') as template_file:
 		options = ''
 		for line in template_file:
 			if 'ssaraopt' in line:
 				options = line.strip('\n').rstrip().split("= ")[1]
 				break;
-					
+
 	# Compute SSARA options to use
 	options = options.split(' ')
 
-	ssara_options = ['ssara_federated_query.py'] + options + ['--print']	
-		
+	ssara_options = ['ssara_federated_query.py'] + options + ['--print']
+
 	return ssara_options
-		
-"""
-    Reads the most recently stored date for the given dataset from the stored_date.date file, and parses the newest data date from ssara_federated_query.
-    Parameters: ssara_output: str, string output from ssara_federated_query.py ... --print
-    Returns:    none
-"""
+
+
 def set_dates(ssara_output):
+	""" Reads the most recently stored date for the given dataset from the stored_date.date file, and parses the newest
+		data date from ssara_federated_query.
+
+		Parameters: ssara_output: str, string output from ssara_federated_query.py ... --print
+		Returns:    none
+	"""
 	global stored_date, most_recent
-	
+
 	most_recent_data = ssara_output.split("\n")[-2]
 	most_recent = datetime.strptime(most_recent_data.split(",")[3], date_format)
 
+	# Checks if the stored_date.date file exists and created the file+directories if it doesn't
+	filename = os.getenv('OPERATIONS') + '/stored_date.date'
+	if not os.path.exists(filename):
+		if not os.path.exists(os.path.dirname(filename)):
+			os.makedirs(os.path.dirname(filename))
+		subprocess.Popen("touch "+filename, shell=True).wait()
+
 	# Write Most Recent Date to File
-	with open(os.getenv('OPERATIONS')+'/stored_date.date', 'rb') as stored_date_file:
-	
+	with open(filename, 'rb') as stored_date_file:
+
 		try:
-			date_line = subprocess.check_output(['grep', dataset, os.getenv('OPERATIONS')+'/stored_date.date']).decode('utf-8')
+			date_line = subprocess.check_output(
+				['grep', dataset, os.getenv('OPERATIONS') + '/stored_date.date']).decode('utf-8')
 			stored_date = datetime.strptime(date_line.split(": ")[1].strip('\n'), date_format)
 		except subprocess.CalledProcessError as e:
-			
+
 			stored_date = datetime.strptime("1970-01-01T12:00:00.000000", date_format)
-			
-			with open(os.getenv('OPERATIONS')+'/stored_date.date', 'a+') as date_file:
-				data = str(dataset + ": "+str(datetime.strftime(most_recent, date_format))+"\n")
+
+			with open(os.getenv('OPERATIONS') + '/stored_date.date', 'a+') as date_file:
+				data = str(dataset + ": " + str(datetime.strftime(most_recent, date_format)) + "\n")
 				date_file.write(data)
 
-"""
-    Compares the most recent and stored dates 
-    Parameters: none
-    Returns:    boolean, whether the most recent data is more recent than the stored date
-"""
+
 def compare_dates():
```

Comment:
Can you refactor out this function? It is only used in one spot and is just as readable if it were inline